identity-iota-core/src/error.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+pub type Result<T, E = Error> = core::result::Result<T, E>;
+
+#[derive(Debug, thiserror::Error, strum::IntoStaticStr)]
+pub enum Error {
+  #[error("{0}")]
+  CoreError(#[from] identity_core::Error),
+  #[error("{0}")]
+  DiffError(#[from] identity_core::diff::Error),
+  #[error("{0}")]
+  InvalidDID(#[from] identity_did::did::DIDError),
+  #[error("{0}")]
+  InvalidDoc(#[from] identity_did::Error),
```

Comment:
This might be opportunity to implement the new error handling guidelines and add new enum variants that do not simply bubble up errors from sub-crates without context. Completely optional for this PR since it is rather bulky though and we would like to prevent merge conflicts.

Reply:
Made some improvements, defer to another PR to prevent too many changes.